src/main/Main.hs

```diff
@@ -283,16 +283,9 @@ commandLineHandler progName isInterpreter = complicatedOptions
                     updateCmd
                     (pure ())
         addCommand' "upgrade"
-                    "Upgrade to the latest stack (experimental)"
+                    "Upgrade to the latest stack"
                     upgradeCmd
-                    ((,) <$> switch
-                              ( long "git"
-                             <> help "Clone from Git instead of downloading from Hackage (more dangerous)" )
-                         <*> strOption
-                              ( long "git-repo"
-                             <> help "Clone from specified git repository"
-                             <> value "https://github.com/commercialhaskell/stack"
-                             <> showDefault ))
+                    (upgradeOpts Nothing) -- FIXME insert the default platform to use
```

Comment:
We already have logic to determine it for downloading GHC, so that could be re-used.

An advantage of passing it in at compile time would be that we could ensure that _exactly_ the same platform is used for the downloaded version as is currently running.  On the other hand, it may sometimes make sense to update the logic for determining the platform (e.g. when new distro versions are released that change shared library versions such as GMP), in which case having it be "static" would be a disadvantage.